### PR TITLE
extension: Limit more API calls

### PR DESCRIPTION
**Pull request by hbons**

### CODE REVIEW

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs [hunk 1]
```diff
@@ -127,16 +129,19 @@ public void GenerateDocuments ()
...
 			};
-			bundle.Manifest = new FigmaManifest () {
+
+			bundle.Manifest = new FigmaManifest() {
 				ApiVersion = AppContext.Current.Version,
-				RemoteApiVersion = AppContext.Api.Version.ToString (),
+				RemoteApiVersion = AppContext.Api.Version.ToString(),
 				Date = DateTime.Now,
-				FileId = fileId
+				FileId = fileProvider.Response.document.id,
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs]: fileProvider.Response could be null here

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs [hunk 2]
```diff
@@ -127,16 +129,19 @@ public void GenerateDocuments ()
...
 
-		public static FigmaBundle Create (string fileId, string directoryPath)
+		public static FigmaBundle Create (IFigmaFileProvider fileProvider, string directoryPath)
 		{
 			var bundle = new FigmaBundle () {
 				DirectoryPath = directoryPath
 			};
-			bundle.Manifest = new FigmaManifest () {
+
+			bundle.Manifest = new FigmaManifest() {
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs]: fileProvider.Response could be null on failed api responses

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs [hunk 3]
```diff
@@ -23,7 +25,7 @@ public class FigmaBundle
 		public string ResourcesDirectoryPath => Path.Combine (DirectoryPath, ResourcesDirectoryName);
 		public string ManifestFilePath => Path.Combine (DirectoryPath, ManifestFileName);
 
-		public string Namespace { get; set; } = "FigmaSharp";
+		public string Namespace { get; set; } = "Figma";
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs]: Why?

Comment (hbons) [reply FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs]: It looks better: `new Figma.myWindow();` Also could clash with our FigmaSharp namespace.

Comment (netonjm) [reply FigmaSharp.NativeControls/FigmaSharp.NativeControls/FigmaBundle.cs]: sounds good!

File: FigmaSharp.Tools/MonoDevelop.Figma/Views/FigmaBundleWindow.cs [hunk 4]
```diff
@@ -86,19 +86,20 @@ private void ItemsRefreshState_Handler (object sender, EventArgs e)
 
 		private async void FigmaUrlTextField_Changed (object sender, EventArgs e)
 		{
+			if (FigmaApiHelper.TryParseFileUrl (FileId, out string fileId)) {
```
Comment (netonjm) [inline FigmaSharp.Tools/MonoDevelop.Figma/Views/FigmaBundleWindow.cs]: not important but clear:

if (!FigmaApiHelper.TryParseFileUrl (FileId, out string fileId)) {
    return;
}

FigmaUrlTextField.StringValue = fileId;

**Comment by netonjm** [comment]
It was implemented in last PR! thank you!

**Pull request closed without merging**